src/appleseed-max-impl/oslutils.cpp

```diff
@@ -143,6 +147,48 @@ asr::ParamArray get_uv_params(Texmap* texmap)
     return uv_params;
 }
 
+
+asr::ParamArray get_output_params(Texmap* texmap)
+{
+    asr::ParamArray output_params;
+    output_params.insert("in_multiplier", fmt_osl_expr(1.0f));
+    output_params.insert("in_clamp_output", fmt_osl_expr(0));
+    output_params.insert("in_invert", fmt_osl_expr(0));
+    output_params.insert("in_colorGain", fmt_osl_expr(foundation::Color3f(1.0f)));
+    output_params.insert("in_colorOffset", fmt_osl_expr(foundation::Color3f(0.0f)));
+    output_params.insert("in_alphaGain", fmt_osl_expr(1.0f));
+    output_params.insert("in_alphaOffset", fmt_osl_expr(0.0f));
+    output_params.insert("in_alphaIsLuminance", fmt_osl_expr(0));
+
+    if (texmap == nullptr)
+        return output_params;
+
+    StdTexoutGen* std_tex_output = nullptr;
+    for (int i = 0, e = texmap->NumRefs(); i < e, std_tex_output == nullptr; ++i)
+    {
+        ReferenceTarget* ref = texmap->GetReference(i);
+        if (ref != nullptr && ref->SuperClassID() == TEXOUTPUT_CLASS_ID)
+        {
+            std_tex_output = dynamic_cast<StdTexoutGen*>(ref);
```

Comment:
So we can't `break` here?

Reply:
I used `; i < e, std_tex_output == nullptr;` in the for loop instead of break here. I thought it'll work. Is it bad practice? It won't work you think?

Reply:
Oh, I missed that. It will work but it's a bit surprising. I would recommend just adding a `break` statement.